go/common/workerpool/workerpool.go

```diff
@@ -4,15 +4,80 @@ package workerpool
 import (
 	"fmt"
 	"sync"
+	"time"
 
 	"github.com/eapache/channels"
 
 	"github.com/oasisprotocol/oasis-core/go/common/logging"
 )
 
+type backoff interface {
```

Comment:
Could we use [`github.com/cenkalti/backoff/v4`](https://pkg.go.dev/github.com/cenkalti/backoff/v4) (which also adds jitter) instead of rolling another backoff implementation?

Reply:
Yeah i think i can use it nicely within the 'expBackoff' struct to avoid manually computing the backoff. Will update.

But i do need to keep the interface/wrapper struct, because i want to share the same backoff between all workers (and cannot do that with plain upstream backoff as it doesn't expose a method to get the current backoff without also incrementing the failure count).

Reply:
Updated